### ISSUE TITLE
how to update the memory bank?  thanks

### ISSUE THREAD

**Issue by shenxiaochenn** [maple-research-lab/AdCo#17]
In fact, I read your code carefully, but I didn't understand the part about updating the memory bank. Specifically, in your code https://github.com/maple-research-lab/AdCo/blob/main/training/train.py  , 60-67 lines, I'm confused. I don't really understand what the g, v, w stands for, so I hope you can give me an answer in your busy schedule. Sincerely thank you~

**Comment by wang3702** [comment]
Thank you for your interest in our work! We also have the following work CaCo with better results: https://github.com/maple-research-lab/CaCo

For your question, please check https://github.com/maple-research-lab/AdCo/issues/8. I hope this issue can help you resolve your confusion. 

W is the actual parameter for the memory bank.

v is a temp variable used to mimic SGD to achieve the momentum update.  Please check here for more details: https://pytorch.org/docs/stable/generated/torch.optim.SGD.html

g is the actual gradient calculated in this step.

**Comment by shenxiaochenn** [comment]
Sincerely thank you for your reply, but I still have doubts. I would like to say a few words about my understanding that in https://github.com/maple-research-lab/AdCo/blob/main/model/AdCo.py ,  176-177 lines,  the normalized W is the memory bank, i.e.,  the negative samples. So how to understand  ‘W is the actual parameter for the memory bank.’? I mean, shouldn't W be a negative sample? Why is it a parameter?

**Comment by wang3702** [comment]
Yes. W are negative samples. However, our negative embeddings W are directly learnable during training. Therefore, they are also trainable parameters.
